pennylane/devices/default_qubit.py

```diff
@@ -888,3 +888,111 @@ def analytic_probability(self, wires=None):
+            seed (Union[int, None]): If provided, it is used to seed the random
+                number generation for generating the Pauli measurements.
+
+        Returns:
+            tensor_like[int]: A tensor with shape (2, T, n), where the first row represents
+            the measured bits and the second represents the recipes used.
+        """
+
+        n_qubits = len(self.wires)
+        device_wires = np.array(self.map_wires(wires))
+
+        if seed is not None:
+            # seed the random measurement generation so that recipes
+            # are the same for different executions with the same seed
+            rng = np.random.RandomState(seed)
+            recipes = rng.randint(0, 3, size=(n_snapshots, n_qubits))
+        else:
+            recipes = np.random.randint(0, 3, size=(n_snapshots, n_qubits))
+
+        obs_list = self._stack(
+            [
+                qml.PauliX.compute_matrix(),
+                qml.PauliY.compute_matrix(),
+                qml.PauliZ.compute_matrix(),
+            ]
+        )
+        uni_list = self._stack(
+            [
+                qml.Hadamard.compute_matrix(),
+                qml.Hadamard.compute_matrix() @ qml.RZ.compute_matrix(-np.pi / 2),
+                qml.Identity.compute_matrix(),
+            ]
+        )
+        obs = obs_list[recipes]
+        uni = uni_list[recipes]
+
+        outcomes = np.zeros((n_snapshots, n_qubits))
+        stacked_state = self._stack([self._state for _ in range(n_snapshots)])
+
+        # There's a significant speedup if we use the following iterative
+        # process to perform the randomized Pauli measurements:
+        #   1. Randomly generate Pauli observables for all snapshots for
+        #      a single qubit (e.g. the first qubit).
+        #   2. Compute the expectation of each Pauli observable on the first
+        #      qubit by tracing out all other qubits.
+        #   3. Sample the first qubit based on each Pauli expectation.
+        #   4. For all snapshots, determine the collapsed state of the remaining
+        #      qubits based on the sample result.
+        #   4. Repeat iteratively until no qubits are remaining.
+        #
+        # Observe that after the first iteration, the second qubit will become the
+        # "first" qubit in the process. The advantage to this approach as opposed to
+        # simulataneously computing the Pauli expectations for each qubit is that
+        # the partial traces are computed over iteratively smaller subsystems, leading
+        # to a significant speed-up.
+
+        for i in range(n_qubits):
+
+            # trace out every qubit except the first
+            first_qubit_state = self._einsum(
+                f"{ABC[n_qubits - i + 1]}{ABC[:n_qubits - i]},{ABC[n_qubits - i + 1]}{ABC[n_qubits - i]}{ABC[1:n_qubits - i]}"
+                f"->{ABC[n_qubits - i + 1]}a{ABC[n_qubits - i]}",
+                stacked_state,
+                self._conj(stacked_state),
+            )
```

Comment:
Since einsum is being used here, does this result in a maximum of 26 qubits being supported? If so, it might make sense to add this as a note box in the docstring (for both here and `qml.classical_shadows`)

Reply:
I believe both lower and uppercase letters are allowed in einsum, so 52 indices in total. I just tried it and this works well into the mid-thirties (at which point my laptop runs out of memory).

Reply:
Ah, I didn't know that! 52 should be good for now, but I still wonder if we should make it explicit so we don't forget lol